### ISSUE TITLE
CI does not run networkd-test.py which is now failing with v239

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#9449]
I noticed that the autopkgtest for networkd fails on Debian:
https://ci.debian.net/data/autopkgtest/testing/amd64/s/systemd/518796/log.gz
```
autopkgtest [16:16:46]: test networkd-test.py: [-----------------------
test_bridge_init (__main__.BridgeTest) ... ok
test_bridge_port_priority (__main__.BridgeTest) ... ok
test_bridge_port_priority_set_zero (__main__.BridgeTest)
It should be possible to set the bridge port priority to 0 ... ok
test_bridge_port_property (__main__.BridgeTest)
Test the "[Bridge]" section keys ... ERROR
test_coldplug_dhcp_ip4_only (__main__.DnsmasqClientTest) ... ERROR
test_coldplug_dhcp_ip4_only_no_ra (__main__.DnsmasqClientTest) ... ERROR
test_coldplug_dhcp_ip6 (__main__.DnsmasqClientTest) ... ERROR
test_coldplug_dhcp_yes_ip4 (__main__.DnsmasqClientTest) ... ERROR
test_coldplug_dhcp_yes_ip4_no_ra (__main__.DnsmasqClientTest) ... ERROR
test_hotplug_dhcp_ip4 (__main__.DnsmasqClientTest) ... Cannot find device "router_eth42"
ERROR
ERROR
test_hotplug_dhcp_ip6 (__main__.DnsmasqClientTest) ... Cannot find device "router_eth42"
ERROR
ERROR
test_resolved_domain_restricted_dns (__main__.DnsmasqClientTest)
resolved: domain-restricted DNS servers ... ERROR
test_resolved_etc_hosts (__main__.DnsmasqClientTest)
resolved queries to /etc/hosts ... ERROR
test_route_only_dns (__main__.DnsmasqClientTest) ... ERROR
test_route_only_dns_all_domains (__main__.DnsmasqClientTest) ... ERROR
test_transient_hostname (__main__.DnsmasqClientTest)
networkd sets transient hostname from DHCP ... ERROR
test_transient_hostname_with_static (__main__.DnsmasqClientTest)
transient hostname is not applied if static hostname exists ... ERROR
test_basic_matching (__main__.MatchClientTest)
Verify the Name= line works throughout this class. ... ERROR
test_inverted_matching (__main__.MatchClientTest)
Verify that a '!'-prefixed value inverts the match. ... ERROR
test_coldplug_dhcp_ip4_only (__main__.NetworkdClientTest) ... ok
test_coldplug_dhcp_ip4_only_no_ra (__main__.NetworkdClientTest) ... ok
test_coldplug_dhcp_ip6 (__main__.NetworkdClientTest) ... skipped 'networkd does not have DHCPv6 server support'
test_coldplug_dhcp_yes_ip4 (__main__.NetworkdClientTest) ... ok
test_coldplug_dhcp_yes_ip4_no_ra (__main__.NetworkdClientTest) ... ok
test_dhcp_timezone (__main__.NetworkdClientTest)
networkd sets time zone from DHCP ... ok
test_dropin (__main__.NetworkdClientTest) ... ok
test_hotplug_dhcp_ip4 (__main__.NetworkdClientTest) ... ok
test_hotplug_dhcp_ip6 (__main__.NetworkdClientTest) ... skipped 'networkd does not have DHCPv6 server support'
test_route_only_dns (__main__.NetworkdClientTest) ... ok
test_route_only_dns_all_domains (__main__.NetworkdClientTest) ... ok
test_search_domains (__main__.NetworkdClientTest) ... ok
test_search_domains_too_long (__main__.NetworkdClientTest) ... ok
test_catchall_config (__main__.UnmanagedClientTest)
Verify link states with a catch-all config, hot-plug. ... ok
test_catchall_config_coldplug (__main__.UnmanagedClientTest)
Verify link states with a catch-all config, cold-plug. ... ok
test_unmanaged_setting (__main__.UnmanagedClientTest)
Verify link states with Unmanaged= settings, hot-plug. ... ok
test_unmanaged_setting_coldplug (__main__.UnmanagedClientTest)
Verify link states with Unmanaged= settings, cold-plug. ... ok

======================================================================
ERROR: test_bridge_port_property (__main__.BridgeTest)
Test the "[Bridge]" section keys
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 175, in setUp
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_coldplug_dhcp_ip4_only (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 400, in test_coldplug_dhcp_ip4_only
    online_timeout=15)
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 307, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_coldplug_dhcp_ip4_only_no_ra (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 405, in test_coldplug_dhcp_ip4_only_no_ra
    extra_opts='IPv6AcceptRA=False')
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 307, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_coldplug_dhcp_ip6 (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 408, in test_coldplug_dhcp_ip6
    self.do_test(coldplug=True, ipv6=True)
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 307, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_coldplug_dhcp_yes_ip4 (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 390, in test_coldplug_dhcp_yes_ip4
    self.do_test(coldplug=True, ipv6=False, online_timeout=15)
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 307, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_coldplug_dhcp_yes_ip4_no_ra (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 395, in test_coldplug_dhcp_yes_ip4_no_ra
    extra_opts='IPv6AcceptRA=False')
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 307, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_hotplug_dhcp_ip4 (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 412, in test_hotplug_dhcp_ip4
    self.do_test(coldplug=False, ipv6=False, online_timeout=15)
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 310, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_hotplug_dhcp_ip4 (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 261, in tearDown
    self.shutdown_iface()
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 514, in shutdown_iface
    subprocess.check_call(['ip', 'link', 'del', 'dev', self.if_router])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ip', 'link', 'del', 'dev', 'router_eth42']' returned non-zero exit status 1.

======================================================================
ERROR: test_hotplug_dhcp_ip6 (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 415, in test_hotplug_dhcp_ip6
    self.do_test(coldplug=False, ipv6=True)
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 310, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_hotplug_dhcp_ip6 (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 261, in tearDown
    self.shutdown_iface()
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 514, in shutdown_iface
    subprocess.check_call(['ip', 'link', 'del', 'dev', self.if_router])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ip', 'link', 'del', 'dev', 'router_eth42']' returned non-zero exit status 1.

======================================================================
ERROR: test_resolved_domain_restricted_dns (__main__.DnsmasqClientTest)
resolved: domain-restricted DNS servers
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 566, in test_resolved_domain_restricted_dns
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_resolved_etc_hosts (__main__.DnsmasqClientTest)
resolved queries to /etc/hosts
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 624, in test_resolved_etc_hosts
    self.do_test(coldplug=None, ipv6=True)
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 314, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_route_only_dns (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 432, in test_route_only_dns
    extra_opts='IPv6AcceptRouterAdvertisements=False')
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 307, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_route_only_dns_all_domains (__main__.DnsmasqClientTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 456, in test_route_only_dns_all_domains
    extra_opts='IPv6AcceptRouterAdvertisements=False')
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 307, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_transient_hostname (__main__.DnsmasqClientTest)
networkd sets transient hostname from DHCP
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 666, in test_transient_hostname
    self.do_test(coldplug=None, extra_opts='IPv6AcceptRA=False', dhcp_mode='ipv4')
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 314, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_transient_hostname_with_static (__main__.DnsmasqClientTest)
transient hostname is not applied if static hostname exists
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 701, in test_transient_hostname_with_static
    self.do_test(coldplug=None, extra_opts='IPv6AcceptRA=False', dhcp_mode='ipv4')
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 314, in do_test
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_basic_matching (__main__.MatchClientTest)
Verify the Name= line works throughout this class.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 942, in test_basic_matching
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

======================================================================
ERROR: test_inverted_matching (__main__.MatchClientTest)
Verify that a '!'-prefixed value inverts the match.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest-lxc.256qx1wa/downtmp/build.Br5/src/test/networkd-test.py", line 957, in test_inverted_matching
    subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
  File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'start', 'systemd-networkd']' returned non-zero exit status 1.

----------------------------------------------------------------------
Ran 36 tests in 115.763s

FAILED (errors=18, skipped=2)
autopkgtest [16:18:43]: test networkd-test.py: -----------------------]
autopkgtest [16:18:43]: test networkd-test.py:  - - - - - - - - - - results - - - - - - - - - -
networkd-test.py     FAIL non-zero exit status 1
autopkgtest [16:18:43]: test networkd-test.py:  - - - - - - - - - - stderr - - - - - - - - - -
Cannot find device "router_eth42"
Cannot find device "router_eth42"
```

The problematic part is, that this test is never run on the upstream / Ubuntu CI afaics. Looking at one of the recent logs, I see:

```
autopkgtest [14:20:16]: test networkd-test.py: [-----------------------
skipped 'networkd is already active'
----------------------------------------------------------------------
Ran 0 tests in 0.012s
OK (skipped=1)
autopkgtest [14:20:16]: test networkd-test.py: -----------------------]
autopkgtest [14:20:17]: test networkd-test.py:  - - - - - - - - - - results - - - - - - - - - -
networkd-test.py     PASS
```

**Comment by mbiebl** [comment]
I guess the failing networkd-test.py in v239 is due to the fact, that networkd was locked down further compared to v238, where it still passed on Debian.

**Comment by yuwata** [comment]
Does the static user and group systemd-network exist? I guess that networkd-test.py requires the static user and group.

**Comment by mbiebl** [comment]
Doesn't seem to make a difference whether I create a static systemd-network user/group or not. It always fails the same.

**Comment by mbiebl** [comment]
@yuwata I can reproduce the failures in a Debian sid qemu VM. So in case you want me to run further diagnostics, just let me know.

**Comment by yuwata** [comment]
@mbiebl Thanks. But unfortunately I do not have any idea now. I will try to investigate the error.

**Comment by filbranden** [comment]
It looks like #9486 at least partially addresses this one...

Is there still a problem to fix here?

**Comment by mbiebl** [comment]
What really fixed it is c4a0a2d5f81ff1fa097be0b431eeaac96cbbb191 afaics, as this allows networkd-test.py to run, even if it's already running (like on Ubuntu 18.04).
